### PR TITLE
Add Avali mutations mod

### PR DESCRIPTION
**Pull request by Arokha**
#### Summary
Mods "Adds 'Avali' mod"

#### Purpose of change
Avali were a race/species/whatever created originally for Starbound and became quite popular in certain circles. This mod adds a mutation tree for representing them in game. Feature-wise, it's a combination of bird and raptor existing mutations plus a few additional custom ones. They play like glass cannons with worse eyesight than birds but far better hearing and fur instead of feathers, but with feathered arms from raptors. The ears mutation is probably the most unique one that's new/custom.

#### Describe the solution
I made this mod for myself and I've been playing with it, so no hard feelings/no wasted work if this is undesirable and is just closed outright because 'oh no too furry' or something. It took like 60 seconds to just make a commit and PR for it.

The species itself is the creation of RyujinZERO who has licensed it as CC BY-SA, which is the same as this game, so I don't see there being a licensing conflict. Especially considering it's just a mod rather than base game content.

#### Describe alternatives you've considered
Not PR'ing this because of hesitation about the reaction to it.

#### Testing
I've been playing with it, and it's almost entirely a remix of base game mutations into a new category, with only a few added.

#### Additional context
The public statement by the original creator licensing them as CC BY-SA: https://www.furaffinity.net/journal/9954644/

### CODE REVIEW

File: data/mods/Avali/mutations/avali_mutations.json [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    "hearing_modifier": 2.5,
+    "threshreq": [ "THRESH_AVALI" ]
+  },
+  {
+    "type": "mutation",
+    "id": "AVI_BLOOD",
+    "name": { "str": "Avali Metabolism" },
+    "points": 0,
+    "mixed_effect": true,
+    "description": "Your muscle response is dependent on ambient temperatures.  You lose 1% of your speed for every 5 (2.8) degrees above 65 F (18.3 C). Your metabolism is more effective, however.",
```
Comment (Night-Pryanik) [inline data/mods/Avali/mutations/avali_mutations.json:80]:
```suggestion
    "description": "Your muscle response is dependent on ambient temperatures.  You lose 1% of your speed for every 5 (2.8 C) degrees above 65 F (18.3 C). Your metabolism is more effective, however.",
```

File: data/mods/Avali/mutations/avali_mutations.json [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    "category": [ "AVALI" ]
+  },
+  {
+    "type": "mutation",
+    "id": "AVI_EARS2",
+    "name": { "str": "Avali Ears" },
+    "points": 2,
+    "visibility": 6,
+    "ugliness": 2,
+    "description": "You have four ears instead of two, and they wiggle and twist towards sounds. Your ability to detect distant noises is quite impressive.",
```
Comment (Night-Pryanik) [inline data/mods/Avali/mutations/avali_mutations.json:67]:
```suggestion
    "description": "You have four ears instead of two, and they wiggle and twist towards sounds.  Your ability to detect distant noises is quite impressive.",
```

**Comment by Night-Pryanik** [comment]
Why not add it to base game instead of a mod?

File: data/mods/Avali/mutations/avali_mutations.json (hunk 1 above)
Comment (Arokha) [reply data/mods/Avali/mutations/avali_mutations.json]: This typo is present in the base game, though I won't fix it in this PR, just putting this here as a reminder. I'll fix this mod's typo obviously though!

**Comment by Arokha** [comment]
> Why not add it to base game instead of a mod?

tl;dr anxiety

Because I am old, and I grew up in a time where mentioning anything with vaguely anthropomorphic characters and roleplaying would get you mocked, so contributing it here felt 'safer' to propose as a mod, so I don't generate drama or something by people saying 'oh no there's weird space bird raptor things in my videogame and I can't turn them off'.

File: data/mods/Avali/mutations/avali_mutations.json [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    "hearing_modifier": 2.5,
+    "threshreq": [ "THRESH_AVALI" ]
+  },
+  {
+    "type": "mutation",
+    "id": "AVI_BLOOD",
+    "name": { "str": "Avali Metabolism" },
+    "points": 0,
+    "mixed_effect": true,
+    "description": "Your muscle response is dependent on ambient temperatures.  You lose 1% of your speed for every 5 (2.8 C) degrees above 65 F (18.3 C). Your metabolism is more effective, however.",
```
Comment (Night-Pryanik) [inline data/mods/Avali/mutations/avali_mutations.json:80]:
```suggestion
    "description": "Your muscle response is dependent on ambient temperatures.  You lose 1% of your speed for every 5 (2.8) degrees above 65 F (18.3 C). Your metabolism is more effective, however.",
```

Sorry that was wrong.

Comment (Arokha) [reply data/mods/Avali/mutations/avali_mutations.json:80]: Why wrong? It seems better with the C in there to me.

**Comment by AtomicFox556** [comment]
> Why not add it to base game instead of a mod?

Personally I wouldn't mind having a lot of very specific mutation trees, but wouldn't that screw up the mutation tree weights, making it impossible to cross certain mutation thresholds? There was a mutation rework recently and I'm not sure if it still works like that, but it was a potential issue before, I think.

Either way you'd probably want it to be called differently in a base game; would seem weird to have many mutation trees literally be called after some specific fictional creatures; not everyone knows what all of them are.

File: data/mods/Avali/mutations/avali_mutations.json (hunk 3 above)
Comment (Night-Pryanik) [reply data/mods/Avali/mutations/avali_mutations.json:80]: Either
`every 5 F (2.8 C) degrees above 65 F (18.3 C)`
or
`every 5 (2.8) degrees above 65 F (18.3 C) respectively`

Comment (Arokha) [reply data/mods/Avali/mutations/avali_mutations.json:80]: I like the former. The 'respectively' might make people have to backtrack to match the pairs. I'll leave it with the former styling.

**Comment by Arokha** [comment]
> Either way you'd probably want it to be called differently in a base game; would seem weird to have many mutation trees literally be called after some specific fictional creatures; not everyone knows what all of them are.

Yeah, fair, so I'll just leave it as a mod.

**Comment by Arokha** [comment]
Not sure what would cause the GCC 9, Curses, LTO build to fail... something about overmap terrain generation? Not sure if that was me or just a random thing.

**Comment by Night-Pryanik** [comment]
This particular error isn't caused by your PR.

**Comment by kevingranade** [comment]
This doesn't fit in mainline, and I'm not particularly inclined to have such a narrowly single topic mod bundled either.
